clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -713,7 +803,8 @@ void LoweringPreparePass::runOnOperation() {
   op->walk([&](Operation *op) {
     if (isa<CmpThreeWayOp, VAArgOp, GlobalOp, DynamicCastOp, StdFindOp,
             IterEndOp, IterBeginOp, ArrayCtor, ArrayDtor, mlir::cir::FuncOp,
-            FModOp, PowOp>(op))
+            CosOp, ExpOp, Exp2Op, LogOp, Log10Op, Log2Op, SinOp, SqrtOp, FModOp,
+            PowOp>(op))
```

Comment:
The chain of `if`s is getting a bit bigger in `runOnOp`. How about:
- We track these in another small vec `mathOpsToTransform`.
- `runOnOp` calls only one function to handle them. Instead of this function dispatching to many `lowerUnaryFPToFPBuiltinOp` calls, it could do a table lookup to pick the strings: ` { {"cosf", "cos", "cosl"}, .... }`, etc.

This should help us pave the road for some later tablegen based approach.